solver/src/analytics.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{metrics::SolverMetrics, settlement::Settlement};
+use bigdecimal::{ToPrimitive, Zero};
+use ethcontract::H160;
+use model::order::OrderUid;
+use num::BigRational;
+use shared::conversions::U256Ext;
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
+
+#[derive(Clone)]
+struct SurplusInfo {
+    solver_name: &'static str,
+    absolute: BigRational,
+    ratio: BigRational,
+}
+
```

Comment:
nit: I noticed that there was a subtle bug because we had two separate comparisons of `SurplusInfo` that did it in different ways. Maybe we can make a helper function?

```rust
impl SurplusInfo {
    fn is_better_than(&self, other: &Self) -> bool;
}
```

In theory, we could also use `impl PartialOrd for SurplusInfo`, but I'm not a fan of that because you get into weird things where the surplus ratio is the same but the solver name is different.